messaging/src/test/java/org/axonframework/common/infra/FilesystemComponentDescriptorTest.java

```diff
@@ -0,0 +1,276 @@
+        // given
+        var component1 = new CircularComponent("Component1");
+        var component2 = new CircularComponent("Component2");
+        component1.setReference(component2);
+        component2.setReference(component1);
+        testSubject.describeProperty("circularRef", component1);
+
+        // when
+        var result = testSubject.describe();
+
+        // then
+        var expected = """
+                /
+                └── circularRef/
+                    ├── _type: CircularComponent
+                    ├── name: Component1
+                    └── reference/
+                        ├── _type: CircularComponent
+                        ├── name: Component2
+                        └── reference -> /circularRef
+                
+                """;
+        assertEquals(normalizeLineEndings(expected), normalizeLineEndings(result));
+    }
+
+    @Test
+    void describeNestedComponentWithCircularReferences() {
+        // given
+        var component1 = new CircularComponent("Component1");
+        var component2 = new CircularComponent("Component2");
+        component1.setReference(component2);
+        component2.setReference(component1);
+
+        var container = new ContainerComponent("Container", component1,
+                                               List.of(component2, component1),
+                                               Map.of("key1", component1, "key2", component2));
+
+        testSubject.describeProperty("container", container);
+
+        // when
+        var result = testSubject.describe();
+
+        // then
+        var expected = """
+                /
+                └── container/
+                    ├── _type: ContainerComponent
+                    ├── name: Container
+                    ├── mainComponent/
+                    │   ├── _type: CircularComponent
+                    │   ├── name: Component1
+                    │   └── reference/
+                    │       ├── _type: CircularComponent
+                    │       ├── name: Component2
+                    │       └── reference -> /container/mainComponent
+                    ├── componentList/
+                    │   ├── [0] -> /container/mainComponent/reference
+                    │   └── [1] -> /container/mainComponent
+                    └── componentMap/
+                        ├── key1 -> /container/mainComponent
+                        └── key2 -> /container/mainComponent/reference
+                """;
+        assertEquals(normalizeLineEndings(expected), normalizeLineEndings(result));
+    }
+
+    @Test
+    void describeSelfReferentialComponent() {
+        // given
+        var component = new CircularComponent("SelfRef");
+        component.setReference(component);
+        testSubject.describeProperty("selfRef", component);
+
+        // when
+        var result = testSubject.describe();
+        System.out.println(result);  // Print for visual inspection during test development
```

Comment:
If you added this for testing yourself, that means you can remove it for the PR :-)